impl/src/main/resources/META-INF/resources/jakarta.faces/faces-uncompressed.js

```diff
@@ -1644,70 +1111,67 @@ if (!((faces && faces.specversion && faces.specversion >= 23000 ) &&
-                        target.value = value;
-                    } else {
-                        target.setAttribute(name, value);
-                    }
-                } else { // if it's IE, then quite a bit more work is required
-                    if (name === 'class') {
-                        target.className = value;
-                    } else if (name === "for") {
-                        name = 'htmlFor';
-                        target.setAttribute(name, value, 0);
-                    } else if (name === 'style') {
-                        target.style.setAttribute('cssText', value, 0);
-                    } else if (name.substring(0, 2) === 'on') {
-                        var c = document.body.appendChild(document.createElement('span'));
-                        try {
-                            c.innerHTML = '<span ' + name + '="' + value + '"/>';
-                            target[name] = c.firstChild[name];
-                        } finally {
-                            document.body.removeChild(c);
-                        }
-                    } else if (name === 'dir') {
-                        if (faces.getProjectStage() == 'Development') {
-                            throw new Error("Cannot set 'dir' attribute in IE");
-                        }
-                    } else {
-                        target.setAttribute(name, value, 0);
-                    }
+                if (name === 'value') {
+                    target.value = value;
+                } else {
+                    target.setAttribute(name, value);
                 }
             }
         };
 
         /**
          * Eval the CDATA of the element.
+         * Evaluate the parsed JavaScript code in a global context.
          * @param element to eval
          * @ignore
          */
-        var doEval = function doEval(element) {
-            var evalText = '';
-            var childNodes = element.childNodes;
-            for (var i = 0; i < childNodes.length; i++) {
-                evalText += childNodes[i].nodeValue;
-            }
-            globalEval(evalText);
+        const doEval = function doEval(element) {
+
+            // should be equal to Node.textContent
+            // let src = '';
+            // for ( const node of element.childNodes ) {
+            //     src += node.nodeValue;
+            // }
+
+            // const nodeText = element.textContent;
+
+            // We have to wrap the call in an anon function because of a firefox bug, where this is incorrectly set
+            // We need to explicitly call window.eval because of a Chrome peculiarity
+
+            // const fn = function() {
+            //     window.eval.call(window, src);
+            // };
+            // fn();
+
+            /**
+             * @ignore
+             */
+            // switch to an async IIFE?
+            // TODO: async requires Safari 10... it was released on 2017... it's ok?
```

Comment:
I'd prefer to wait until we can convert to typescript.

Reply:
ok

Reply:
I've spent some time to investigate the overall deprecation of old software...

Actually Safari 10 - 11 - 12 - 13 - 14 are not supported anymore by Apple 

Even a 10 years old Macbook can upgrade to 
"Big Sur", which include Safari version 15.5

https://discussions.apple.com/thread/253968805

IOS 11 with Safari 11 is availble since September 2017 ( 5 years ago )
and it's available for upgrade even of iPhone 5S (which is a 2013 phone)

https://www.wired.co.uk/article/ios-11-features-update-download-release-date-uk#:~:text=Which%20devices%20support%20iOS%2011,2%20and%205th-gen%20iPad

IE11 is official in EOL from Microsoft 

https://docs.microsoft.com/en-us/lifecycle/announcements/internet-explorer-11-end-of-support

So it is more than safe to updagrade to full ES6 support with async/await and Promise 

Even more If you think that at the moment:

1) the are no Official Application Servers supporting Jakarta EE 10
2) there will be no Official Application Servers supporting Jakarta EE 10.1 before March 2023
    (when Faces 4.1 will probably be released) 


A side note on TypeScript

ES2022 is around the corner and is on par with TypeScript

https://dev.to/jasmin/whats-new-in-es2022-1de6

Obviously anyone is free do whatever they want with open source code but to me 
rewrite all that code to TypeScript in 2023 is only a waste of (precious) time which 
could spent to improve Java classes to perform better and cleaner and resolve bugs 

my 2 cents